packages/server/src/bin.ts

```diff
@@ -0,0 +1,41 @@
+import dotenv from 'dotenv'
+import { IpldStore, MemoryBlockstore, PersistentBlockstore } from '@adxp/repo'
+import * as crypto from '@adxp/crypto'
+import Database from './db'
+import server from './index'
+import { ServerConfig } from './config'
+
+const run = async () => {
+  const env = process.env.ENV
+  if (env) {
+    dotenv.config({ path: `./.${env}.env` })
+  } else {
+    dotenv.config()
+  }
+
+  let blockstore: IpldStore
+  let db: Database
+
+  const cfg = ServerConfig.readEnv()
+
+  if (cfg.blockstoreLocation) {
+    blockstore = new PersistentBlockstore(cfg.blockstoreLocation)
+  } else {
+    blockstore = new MemoryBlockstore()
+  }
+
+  if (cfg.databaseLocation) {
+    db = await Database.sqlite(cfg.databaseLocation)
+  } else {
+    db = await Database.memory()
+  }
+
+  const keypair = await crypto.EcdsaKeypair.create()
+
+  const s = server(blockstore, db, keypair, cfg)
```

Comment:
im tempted to just stick this all in `cfg` 🤔

Reply:
I'd be fine with that